pkg/controller/propagator/propagation.go

```diff
@@ -210,6 +201,89 @@ func (r *ReconcilePolicy) handleRootPolicy(instance *policiesv1.Policy) error {
 	return nil
 }
 
+// getApplicationPlacementDecisions return the placement decisions from an application
+// lifecycle placementrule
+func getApplicationPlacementDecisions(c client.Client, pb policiesv1.PlacementBinding, instance *policiesv1.Policy) ([]appsv1.PlacementDecision, *policiesv1.Placement, error) {
+	plr := &appsv1.PlacementRule{}
+	err := c.Get(context.TODO(), types.NamespacedName{Namespace: instance.GetNamespace(),
+		Name: pb.PlacementRef.Name}, plr)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Failed to get plr...", "Namespace", instance.GetNamespace(), "Name",
+			pb.PlacementRef.Name)
+		return nil, nil, err
+	}
+	// plr found, add current plcmnt to placement
+	placement := &policiesv1.Placement{
+		PlacementBinding: pb.GetName(),
+		PlacementRule:    plr.GetName(),
+		// Decisions:        plr.Status.Decisions,
+	}
+	return plr.Status.Decisions, placement, nil
+}
+
+// getClusterPlacementDecisions return the placement decisions from cluster
+// placement decisions
+func getClusterPlacementDecisions(c client.Client, pb policiesv1.PlacementBinding, instance *policiesv1.Policy) ([]appsv1.PlacementDecision, *policiesv1.Placement, error) {
+	plr := &clusterv1alpha1.Placement{}
+	err := c.Get(context.TODO(), types.NamespacedName{Namespace: instance.GetNamespace(),
+		Name: pb.PlacementRef.Name}, plr)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Failed to get plr...", "Namespace", instance.GetNamespace(), "Name",
+			pb.PlacementRef.Name)
+		return nil, nil, err
+	}
+	// plr found, add current plcmnt to placement
+	placement := &policiesv1.Placement{
+		PlacementBinding: pb.GetName(),
+		Placement:        plr.GetName(),
+		// Decisions:        plr.Status.Decisions,
+	}
+	list := &clusterv1alpha1.PlacementDecisionList{}
+	lopts := &client.ListOptions{Namespace: instance.GetNamespace()}
+
+	opts := client.MatchingLabels{"cluster.open-cluster-management.io/placement": plr.GetName()}
+	opts.ApplyToList(lopts)
+	err = c.List(context.TODO(), list, lopts)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Failed to get plr...", "Namespace", instance.GetNamespace(), "Name",
+			pb.PlacementRef.Name)
+		return nil, nil, err
+	}
+	var decisions []appsv1.PlacementDecision
+	decisions = make([]appsv1.PlacementDecision, 0, 100)
```

Comment:
Why specify a capacity when the amount is unknown? Is it really common to have 100+ placement decisions returned?

Perhaps, you could start with `len(list.Items)` or `len(list.Items) * 2` or no capacity at all and let Go handle that?

Reply:
Right, I'll take that out and I expect we won't see any performance issues, but if we do then a comment about why a size is there will be added at that time.